### ISSUE TITLE
'click' doesn't fire on a map the second time

### ISSUE THREAD

**Issue by gerrod** [Leaflet/Leaflet#3751]
Hi guys - I think I may have found a bug in leaflet. Basic scenario is that we've created a simple control that lets you select a location, and then places a marker in that location. You can see a cut-down version of this here (which also demonstrates the bug) - https://jsbin.com/joropugotu/edit?js,output. Make sure you drag the output window wide enough to see the white "location selector" button:

![image](https://cloud.githubusercontent.com/assets/865932/9299317/037c826a-44fe-11e5-8d09-ff4fabb13e26.png)

So, the basic premise is as follows:

1. Click the "location selector" button => button activates
2. Now click on the map => a marker drops where you clicked
3. Click again on the "location selector" button => button activates
4. Click again on the map => **the marker's location is not updated**
5. Click on the map one more time => now the marker's location is updated.

It only happens on that "second" (and subsequent) click - the first time the control is used, the location selector works as expected. After this, you have to click **twice** in order to update the location.

I've tried this with the current 1.0.0 beta and it still seems to be a bug (the "selector button" doesn't quite work properly, but once you click it a few times you can see the same behaviour happen).

There's a few issues that sound "similar" but from what I can tell they're mainly about having a popup with a close button destroying the map's click events.

Please let me know if I can supply any further information!

**Comment by jieter** [comment]
Some observations: 
 - If I click the control twice and then click on the map, the marker is only placed after the second click on the map.
 - If I add `map.on('click', function () { console.log('clicked'); });`, it does not log 'clicked' to the console for the first click on the map in the above example.

I think you should try to remove more things from your example to get to the problem, as it is not obvious what exactly is going on now...

**Comment by gerrod** [comment]
Thanks for the reply @jieter - I'll admit that I set up the example to mimic our solution as closely as I could, but I took your suggestion and [created a much cut-down version](https://jsbin.com/cacoyedaca/edit?html,js,output) and sure enough, it works as expected. I'd be surprised then if the bug is indeed a leaflet bug, but I'll keep this issue open until I can diagnose more fully what the problem is. 

Again, thanks for your help.
